.github/workflows/ci.yaml

```diff
@@ -15,7 +15,12 @@ jobs:
       - uses: actions/checkout@v2
       - uses: astral-sh/setup-uv@v5
       - uses: astral-sh/ruff-action@v3
+      - uses: astral-sh/ruff-action@v3
+        with:
+          args: "format --check --diff"
       - name: Mypy
-        run: uv run mypy
+        run: uv run mypy --group dev
```

Comment:
This reduces the number of packages installed when running lints/testing, which can help prevent masking missing imports.